cpp/include/cudf_test/iterator_utilities.hpp

```diff
@@ -97,10 +97,30 @@ static auto iterator_with_null_at(cudf::host_span<cudf::size_type const> const&
  * @param index The index for which the validity iterator must return `false` (i.e. null)
  * @return auto Validity iterator
  */
-static auto iterator_with_null_at(cudf::size_type const& index)
+inline auto iterator_with_null_at(cudf::size_type const& index)
```

Comment:
Hmm. Could you please explain (in a reply here, not in the code) why `inline` removes the `defined but not used` error, but `static` does not?
My guess is that this doesn't have to do with visibility/linkage, since both `static` and `inline` force internal linkage.

Reply:
Maybe this is useful: https://stackoverflow.com/a/2846143

Reply:
Sure.

Reply:
that was fast!

Reply:
> `inline` does not have any linkage---it is just expanded in-place.

Hmm... I think I understand how this works now.

My (likely outdated) understanding is that the `inline` directive is a guideline, and not guaranteed. So if the `inline` is declined, a choice would need to be made about linkage/visibility outside the TU. From the description [here](https://en.cppreference.com/w/cpp/language/inline), it defaults to _external_ linkage, unless it's defined as `static`.

Both `static` and `inline` allow the same function to have multiple definitions across different translation units. 
1. With `static`, there is internal linkage, and the definition local to the TU wins for that TU.
2. With `inline` (and no `static`), there is external linkage. I guess the multiple definitions aren't flagged as errors, but the behaviour is undefined if the definitions don't match.

Thanks, I'll resolve this.